cleanlab/classification.py

```diff
@@ -894,11 +901,16 @@ def find_label_issues(
         labels = labels_to_array(labels)
         if self.verbose:
             print("Using predicted probabilities to identify label issues ...")
-        label_issues_mask = filter.find_label_issues(
-            labels,
-            pred_probs,
-            **self.find_label_issues_kwargs,
-        )
+        if self.low_memory:
```

Comment:
Since `find_label_issues_kwargs` is not being used  when `low_memory=True`, perhaps we want to raise a warning (or print statement) informing users that `find_label_issues_kwargs` is not used in this scenario (tho only raised if `find_label_issues_kwargs != {}`).

`find_label_issues_batched` also has additional kwargs but supporting that could be messy/a little complicated so I think it is not the most important to add, for now I think the warning is good enough

Reply:
https://github.com/cleanlab/cleanlab/blob/0cd40abbb28f4a92b2cea3813b806ebd968cf9ac/cleanlab/classification.py#L785
This function seems to be adding some default keyword arguments like `min_examples_per_class` to `find_label_issues_kwargs`. Does thie mean that the warning should always be shown when `low_memory=True` or is it only applicable for non-default (explicit) kwargs passed via `find_label_issues_kwargs`?

Reply:
Yes the latter is right, warning should be raised when a user explicitly passes in `find_label_issues_kwargs`.
ie. on [line 792](https://github.com/cleanlab/cleanlab/blob/c8ad4fd589f14369cf2cc36eb16b395dcc809c79/cleanlab/classification.py#L792) here you can do something like
```
if low_memory == True and len(self.find_label_issues_kwargs) != 0:
    # print some warning message
else:
    self._process_label_issues_kwargs(self.find_label_issues_kwargs)
```

Reply:
In fact, this method computes many values such as `confident_joint` and more which are eventually added to `find_label_issue_kwargs` to be passed to `filter.find_label_issues`. 

However when `low_memory=True` they are not being used, hence I believe these rows here are completely irrelevant and should not be executed (they are just wasted computes): 
https://github.com/cleanlab/cleanlab/blob/c8ad4fd589f14369cf2cc36eb16b395dcc809c79/cleanlab/classification.py#L792-L899

My suggestion would be to move the `if self.low_memory` check much earlier and skip any computes that are not necessary (should be just the ones I mentioned above, but would be good to double check if I accidentally ignored too many things, or missed any).

After making that change, if you could rerun your runtime and performance benchmarks (if not too much work) and report back that would be super helpful (and of course make sure this doesn't make thing worse than it was before - ML metrics wise it should not change from what you already have). Thanks!

Reply:
That makes sense!  When you say "move the if self.low_memory check much earlier and skip any computes that are not necessary", do you mean something like this? (i.e. throw all computation for `filter.find_label_issues` in else). 

```
if self.low_memory:
  # find_label_issues_batched
else:
   # Compute noise matrix, confident joint etc.
   # filter.find label issues
```

Reply:
Yup that's right! You can place all the code that is unrelated to `find_label_issues_batched` in the else block.

And yes good catch! `pred_probs` is necessary for either case so we would need to make sure that is checked and computed if necessary.

Reply:
Thank you! I have two follow-up questions based on initial testing.

**Regarding `pred_probs`**
If `pred_probs` is not passed and it is calculated before checking `low_memory`, then the following code snippet, where both `noise_matrix` and `inverse_noise_matrix` are not passed and `pred_probs is None`, will never be executed. 
https://github.com/cleanlab/cleanlab/blob/c8ad4fd589f14369cf2cc36eb16b395dcc809c79/cleanlab/classification.py#L820-L842
I should also mention that  at this point, the `pred_probs` calculation is being performed _after_ the above snippet, hence it is not dead code. So if this is **not** the intended consequence, then one solution would be to check and calculate `pred_probs` separately within the `low_memory` condition, and otherwise proceed as before.

```python
if low_memory:
    if pred_probs is None:
           # calculate
else:
    # Follow same steps as before
 ```
 
 **Unexpected failing unit-test**
When I was testing the above-mentioned changes, I noticed that the following unit-test is failing.
https://github.com/cleanlab/cleanlab/blob/c8ad4fd589f14369cf2cc36eb16b395dcc809c79/tests/test_classification.py#L742-L744
It is failing for the following code snippet
https://github.com/cleanlab/cleanlab/blob/0cd40abbb28f4a92b2cea3813b806ebd968cf9ac/cleanlab/classification.py#L837-L850
I believe this is happening because when computing the `confident_joint`, the dataframe to array conversion is not being done properly. 
https://github.com/cleanlab/cleanlab/blob/0cd40abbb28f4a92b2cea3813b806ebd968cf9ac/cleanlab/count.py#L578
It converts the labels to `[[[0], [0], [0], [0], [0]]]` instead of `[0 0 0 0 0 0 0 0 0 0 2]`.
Changing this to `labels = labels_to_array(labels)` seems to pass the test. 

If this makes sense, I can make this change as well (and maybe also add a unit-test for this specific case).

Reply:
Yes you will have to make sure the `pred_probs` are computed in either case (even if you might have some repeated code). There is also this function here that just extracts the `pred_probs` and not the noise matrix and other things: 
https://github.com/cleanlab/cleanlab/blob/master/cleanlab/count.py#L1070

It computes the same number of things under the hood but it might help with storing less information.

I'm not entire sure why the unittest error you mentioned is popping up after the code change? Currently the unittests are passing and I don't believe any changes you make should alter the shape of `labels`? It's tough for me to diagnose as well without seeing the code change/error, so if you're not sure how to resolve it/need my help you can push the failing commits and let me know if you want me to take a look at it.